Дан файл csv с данными по просмотрам фильмов и пользовательским оценкам фильмов.
Необходимо найти:
1.	Общее число оценок в файле
2.	Общее количество пользователей, поставивших оценки
3.	Общее количество оцененных фильмов
4.	ID самого активного пользователя
5.	Фильм, собравший наибольшее количество оценок


Анализ задачи:
Загружаем csv файл.
С использованием встроенных функций библиотеки pandas работаем с колонками таблицы, анализируем их и находим необходимые величины, указанные в пунктах задания.
Общее число оценок - это число колонок в файле.
Количество пользователей - число различных id в колонке пользователей. Его можно определить как длину множества, в которое загружены все id пользователей.

Загружаем csv файл:

In [19]:
import pandas as pd
import numpy as np

file_name = 'ratings_small.csv' # source file
data_table = pd.read_csv("ratings_small.csv")

1.	Общее число оценок в файле - это по факту число строк файла

In [20]:
marks_number = data_table.last_valid_index() + 1
print(f"Marks number: {marks_number}")

Marks number: 100004


2.	Общее количество пользователей, поставивших оценки - число различных id пользователей.

In [21]:
users_number = data_table['userId'].nunique()
print(f"Users number: {users_number}")

Users number: 671


3.	Общее количество оцененных фильмов - число различных идентификаторов фильмов.

In [22]:
films_number = data_table['movieId'].nunique()
print(f"Films number: {films_number}")

Films number: 9066


4.	ID самого активного пользователя. 
Самый активный пользователь наиболее часто упоминается в колонке с id пользователей.
Поэтому можно применить метод value_counts() к данной колонке, и он даст объект series, который покажет для каждого элемента колонки  количество вхождений этого элемента.
Далее можно отсортировать series по значениям количества вхождений.
Самое большое значение количества вхождений будет стоять рядом с ID самого активного пользователя.

In [23]:
usr = data_table['userId'].value_counts()
usr_sorted = usr.sort_values()
print(usr_sorted)
best_user = usr_sorted.index[-1]
print("The most active user:", best_user)

userId
1        20
657      20
651      20
35       20
76       20
       ... 
73     1610
15     1700
624    1735
564    1868
547    2391
Name: count, Length: 671, dtype: int64
The most active user: 547


5.	Фильм, собравший наибольшее количество оценок.
Наиболее активно оцениваемый фильм наиболее часто упоминается в колонке с id фильмов.
Поэтому можно применить метод value_counts() к данной колонке, и он даст объект series, который покажет для каждого элемента колонки  количество вхождений этого элемента.
Далее можно отсортировать series по значениям количества вхождений.
Самое большое значение количества вхождений будет стоять рядом с ID самого активно оцениваемого фильма.

In [25]:
flm = data_table['movieId'].value_counts()
flm_sorted = flm.sort_values()
print(flm_sorted)
best_film = flm_sorted.index[-1]
print("The most rated film:", best_film)

movieId
6425        1
147845      1
146682      1
143657      1
143377      1
         ... 
260       291
593       304
318       311
296       324
356       341
Name: count, Length: 9066, dtype: int64
The most rated film: 356


Итак, нами был проделан анализ таблицы с оценками фильмов пользователями.

Выводы:
1. Общее число оценок в файле составило 100 004.
2.	Общее количество пользователей, поставивших оценки - 671.
3.	Общее количество оцененных фильмов - 9066. 
4.	ID самого активного пользователя - 547.
5.	Фильм, собравший наибольшее количество оценок - 356.